app/models/comment.rb

```diff
@@ -5,4 +5,24 @@ class Comment < ApplicationRecord
   has_many   :liking_users, through: :comment_liking_relations,
                             source: :liking_user
   validates_presence_of :content
+
+  def points
+    self.comment_liking_relations.count
+  end
+
+  def is_liked_by?(user)
+    !!my_liking_relation(user)
+  end
+
+  def my_liking_relation(user)
+      comment_liking_relations.find_by(liking_user: user) if user
+  end
+
+  def icon_data(current_user)
```

Comment:
to jednak chyba do konca nie pasuje do modelu.

robisz high coupling miedzy tym jak wyglada UI a modelem poprzez klase text-highlight

Reply:
Ech.. Okej. Problemem jest to, że jak jest ikona lajkowania komentarza, to ona jest forms'em, który wysyła POST do Liking Relations Controller.
Ale jeśli zalajkujemy. to muszę przy użyciu turbo-stream podmienić cały forms innym formsem, który ma zmienione dane formsa: POST->DELETE, oraz klasę ikony z na text-highlight, oraz jeszcze jedna pierdoła.

Chciałem mieć jeden partial, by być DRY. Ale do tego potrzebowałem logiki, która określi jakie są te dane do forms'a.

Ale masz rację, że to się robi tak skompllikowane, że to nie ma sensu.

ALBO nie będę DRY i po prostu zrobię 2 partiale podobnych formsów i wtedy nie potrzebuje tego icon_data.
ALBO mogę tę całą logikę zaszyć w controlerze Comment Liking Relation. Ikona będzie forms'em, który będzie aktywował metodę ToggleLike. Metoda "ToggleLike", wykona logikę i albo wykona POST albo DELETE wewnątrz kontrolera.

Reply:
Jednak robię opcję 1. Nie będę DRY i robię 2 partiale.